zetaclient/chains/bitcoin/signer/fee_bumper.go

```diff
@@ -0,0 +1,209 @@
+package signer
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"time"
+
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/btcsuite/btcd/mempool"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog"
+
+	"github.com/zeta-chain/node/pkg/chains"
+	"github.com/zeta-chain/node/pkg/constant"
+	mathpkg "github.com/zeta-chain/node/pkg/math"
+	"github.com/zeta-chain/node/zetaclient/chains/bitcoin/common"
+)
+
+const (
+	// feeRateCap is the maximum average fee rate for CPFP fee bumping
+	// 100 sat/vB is a heuristic based on Bitcoin mempool statistics to avoid excessive fees
+	// see: https://mempool.space/graphs/mempool#3y
+	feeRateCap = 100
+
+	// decentFeeBumpPercent is a decent percentage for a fee rate bump.
+	// The value20% is a heuristic, not mandated by the Bitcoin protocol. It is used to measure the gap between
+	// the old fee rate, new fee rate and live fee rate to emit warning messages during RBF fee bumping.
+	decentFeeBumpPercent = 20
+)
+
+// CPFPFeeBumper is a helper struct to contain CPFP (child-pays-for-parent) fee bumping logic
+type CPFPFeeBumper struct {
+	Ctx context.Context
+
+	Chain chains.Chain
+
+	// RPC is the interface to interact with the Bitcoin chain
+	RPC RPC
+
+	// Tx is the stuck transaction to bump
+	Tx *btcutil.Tx
+
+	// MinRelayFee is the minimum relay fee in BTC
+	MinRelayFee float64
+
+	// CCTXRate is the most recent fee rate of the CCTX
+	CCTXRate int64
+
+	// LiveRate is the most recent market fee rate
+	LiveRate int64
+
+	// TotalTxs is the total number of stuck TSS txs
+	TotalTxs int64
+
+	// TotalFees is the total fees of all stuck TSS txs
+	TotalFees int64
+
+	// TotalVSize is the total vsize of all stuck TSS txs
+	TotalVSize int64
+
+	// AvgFeeRate is the average fee rate of all stuck TSS txs
+	AvgFeeRate int64
+
+	Logger zerolog.Logger
+}
+
+// NewCPFPFeeBumper creates a new CPFPFeeBumper
+func NewCPFPFeeBumper(
+	ctx context.Context,
+	rpc RPC,
+	chain chains.Chain,
+	tx *btcutil.Tx,
+	cctxRate int64,
+	minRelayFee float64,
+	logger zerolog.Logger,
+) (*CPFPFeeBumper, error) {
+	fb := &CPFPFeeBumper{
+		Ctx:         ctx,
+		Chain:       chain,
+		RPC:         rpc,
+		Tx:          tx,
+		MinRelayFee: minRelayFee,
+		CCTXRate:    cctxRate,
+		Logger:      logger,
+	}
+
+	err := fb.FetchFeeBumpInfo()
+	if err != nil {
+		return nil, err
+	}
+	return fb, nil
+}
+
+// BumpTxFee bumps the fee of the stuck transaction using reserved bump fees
+func (b *CPFPFeeBumper) BumpTxFee() (*wire.MsgTx, int64, int64, error) {
```

Comment:
Let's not return more than 3 variables w/o "result" struct

```suggestion
func (b *CPFPFeeBumper) BumpTxFee() (BumpResult, error) {
```